lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -9,11 +9,18 @@ class DeclarationListener
     BASIC_OBJECT_NESTING = T.let(["BasicObject"].freeze, T::Array[String])
 
     sig do
-      params(index: Index, dispatcher: Prism::Dispatcher, parse_result: Prism::ParseResult, file_path: String).void
+      params(
+        index: Index,
+        dispatcher: Prism::Dispatcher,
+        parse_result: Prism::ParseResult,
+        file_path: String,
+        augmentations: T::Array[Augmentation],
+      ).void
     end
-    def initialize(index, dispatcher, parse_result, file_path)
+    def initialize(index, dispatcher, parse_result, file_path, augmentations = [])
```

Comment:
I think `augmentations` can be a keyword argument as I consider it a additional option.